core/chains/evm/client/rpc_client.go

```diff
@@ -412,8 +435,46 @@ func (r *rpcClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) err
 	duration := time.Since(start)
 
 	r.logResult(lggr, err, duration, r.getRPCDomain(), "BatchCallContext")
+	if err != nil {
+		return err
+	}
 
-	return err
+	// populated requested finalized block with correct value
+	if r.chainType == chaintype.ChainAstar {
```

Comment:
We could optimize this to avoid iterating through the whole batch again if we know there was no finalized block requested:
```suggestion
	if r.chainType == chaintype.ChainAstar && astarRawLatestFinalizedBlock != nil {
```